### PR TITLE
Fix UnusedOpensDiagnosticAnalyzer

### PR DESCRIPTION
**Pull request by vasily-kirichenko**
This fixes https://github.com/Microsoft/visualfsharp/issues/3775

### CODE REVIEW

File: vsintegration/src/FSharp.Editor/Diagnostics/UnusedOpensDiagnosticAnalyzer.fs [hunk 1]
```diff
@@ -74,11 +74,9 @@ module private UnusedOpens =
...
 
     let symbolIsFullyQualified (sourceText: SourceText) (sym: FSharpSymbolUse) (fullName: string) =
-        match RoslynHelpers.TryFSharpRangeToTextSpan(sourceText, sym.RangeAlternate) with
-        | Some span // check that the symbol hasn't provided an invalid span
-            when sourceText.Length < span.Start 
-              || sourceText.Length < span.End -> false
-        | Some span -> sourceText.ToString span = fullName
+        let lineStr = sourceText.Lines.[Line.toZ sym.RangeAlternate.StartLine].ToString()
+        match QuickParse.GetCompleteIdentifierIsland true lineStr sym.RangeAlternate.EndColumn with
+        | Some (island, _, _) -> island = fullName
```
Comment (forki) [inline vsintegration/src/FSharp.Editor/Diagnostics/UnusedOpensDiagnosticAnalyzer.fs]: Some island: 

giphy

Comment (vasily-kirichenko) [reply vsintegration/src/FSharp.Editor/Diagnostics/UnusedOpensDiagnosticAnalyzer.fs]: :)

**Comment by vasily-kirichenko** [comment]
It's ready.

**Comment by Krzysztof-Cieslak** [comment]
❤️

**Comment by cartermp** [comment]
I'll test this today. Great change!  Though I think it's a bit bigger than just fixing #3775 😄

**Comment by cartermp** [comment]
Doesn't appear to grey out either unused open for me:
not-greyed

**Comment by Krzysztof-Cieslak** [comment]
You sure it's enabled in your VS? ;P

Latest fixes seems to work for me in Iondie:
image

**Comment by cartermp** [comment]
I'm just going to `git clean -xdfagahgsdl;aksgdo[paejg;ameglajksndgl;kjasdg` my repo and try this again, since it's still not working for me with the installed VSIX from this PR.

However, I did notice that in Blue Theme (Extra Contrast) the grey text is not shown.  Seems like there's a separate bug there.

**Comment by matthid** [comment]
`git clean -xdff` is actually a thing ;)

**Comment by vasily-kirichenko** [comment]
I was thinking to make it a warning, so it’s green squiggled, like in Code. Grayed text is too subtle to notice.

**Comment by cartermp** [comment]
Works!  I just had to `git clean -xdfph'nglui mglw'nafh Cthulhu R'lyeh wgah'nagl fhtagn`

**Comment by saul** [comment]
Looks great - love the tests 😄

**Comment by vasily-kirichenko** [comment]
Closing in favor of https://github.com/Microsoft/visualfsharp/pull/3803

**Pull request closed without merging**